x-pack/plugins/task_manager/server/polling/delay_on_claim_conflicts.test.ts

```diff
@@ -46,15 +46,12 @@ describe('delayOnClaimConflicts', () => {
       const maxWorkers = 10;
       const taskLifecycleEvents$ = new Subject<TaskLifecycleEvent>();
 
-      const delays$ = delayOnClaimConflicts(
-        of(maxWorkers),
-        of(pollInterval),
-        taskLifecycleEvents$,
-        80,
-        2
-      )
-        .pipe(take(2), bufferCount(2))
-        .toPromise<number[]>();
+      const delays$ = firstValueFrom(
+        delayOnClaimConflicts(of(maxWorkers), of(pollInterval), taskLifecycleEvents$, 80, 2).pipe(
+          take(2),
+          bufferCount(2)
+        )
+      );
```

Comment:
All of these refactoring introducing `firstValueFrom` seem to have dropped the typing.
Could we bring those back?
Other than that it looks good to me but I don't want to lose the type checks.